### PR TITLE
Add OCI ci and make tests not silently failing

### PR DESCRIPTION
**Pull request by CarlSchwan**

### CODE REVIEW

**Comment by ArtificialOwl** [comment]
- [ ] verify that Oracle create/migrate database correctly

**Comment by CarlSchwan** [comment]
@ArtificialOwl I ended up disabling the Api tests as they need way too much adaption. I guess they need to be ported away from the Circles v1 API to the `CirclesManager`

The Controller tests now are running again and are passing the ci and it's better than nothing :)

**Comment by ArtificialOwl** [comment]
Issue on php7.40-oci seems to come from the fact some some SQL table field names/aliases are longer than 30 chars

**Comment by ArtificialOwl** [comment]
rebased

**Comment by ArtificialOwl** [comment]
I will work on this, but won't be backported for next minor release

File: lib/Migration/Version0024Date20220203123902.php [hunk 1]
```diff
@@ -124,6 +124,11 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 			$table->setPrimaryKey(['token', 'instance']);
 		}
 
+		if ($schema->hasTable('circles_membership')) {
```
Comment (CarlSchwan) [inline lib/Migration/Version0024Date20220203123902.php:127]: this needs to be moved in another migration :)

Comment (ArtificialOwl) [reply lib/Migration/Version0024Date20220203123902.php]: fixed :]

File: lib/Migration/Version0024Date20220317190331.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+class Version0024Date20220317190331 extends SimpleMigrationStep {
+
+
+	/**
+	 * @param IDBConnection $connection
+	 */
+	public function __construct(IDBConnection $connection) {
+	}
+
```
Comment (CarlSchwan) [inline lib/Migration/Version0024Date20220317190331.php:49]:
```suggestion
```

**Comment by ArtificialOwl** [comment]
rebased, adding a condition that there is no primarykey before creation.

**Review by ArtificialOwl — APPROVED** [review]

**Pull request merged**